fe1-web/components/QRCode.tsx

```diff
@@ -1,23 +1,46 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import TextBlock from 'components/TextBlock';
 import { KeyPairStore } from 'store/stores';
+import QRCodeDisplay from 'qrcode.react';
+import { View } from 'react-native';
+import styleContainer from 'styles/stylesheets/container';
+import TextBlock from './TextBlock';
+
+/**
+ * Creates and displays a QR code with the public key of the current user
+ */
 
 const QRCode = (props: IPropTypes) => {
   const identity = KeyPairStore.getPublicKey().toString();
   const { visibility } = props;
+  const { size } = props;
+  const { text } = props;
 
-  // For now, we simply display a text block with the public key
+  // Displays a QR code with the Public key of the current user
   return (visibility)
-    ? <TextBlock bold text={`ID (pk) :   ${identity}`} />
+    ? (
+      <>
+        <TextBlock text={String(text)} />
+        <View style={[styleContainer.topCenter, { padding: 10 }]}>
+          <QRCodeDisplay value={identity} size={Number(size)} />
+        </View>
+      </>
+    )
     : null;
 };
 
 const propTypes = {
   visibility: PropTypes.bool.isRequired,
+  size: PropTypes.number,
+  text: PropTypes.string,
```

Comment:
Why did you choose to put text as optional? Can you imagine a case where you would display an "empty QRcode"?

Reply:
The text value was about what would be displayed above the QR code, so usually something like: "ID (Public Key):", 
I made it optional because I thought sometimes it would be nice to just display the QR code and no text.
But this is redundant if I remove the `TextBlock` out of the component as you mentioned above.